src/dopri.ts

```diff
@@ -0,0 +1,149 @@
+import * as dopri5 from "./dopri5/stepper";
+import * as interpolator from "./interpolator";
+import {RhsFn, Stepper} from "./types";
+import * as utils from "./utils";
+
+// needed for ES5 - will be ~= Number.EPSILON in ES6
+const DBL_EPSILON = Math.pow(2, -52); // = 2.220446049250313e-16
+const STEP_FACTOR_MIN = 1e-4;
+
+export function integrate(rhs: RhsFn, y: number[],
+                          t0: number, t1: number) {
+    const solver = new Dopri(rhs, y.length);
+    solver.initialise(t0, y);
+    return solver.run(t1);
+}
+
+function integrationError(message: string, t: number) {
+    return new Error(`Integration failure: ${message} at ${t}`);
+}
+
+export class Dopri {
+    public stepper: Stepper;
+    public t: number = 0.0;
+    public h: number = 0.0;
+
+    // state
+    public nSteps: number = 0;
+    public nStepsAccepted: number = 0;
+    public nStepsRejected: number = 0;
+
+    public stiffNStiff: number = 0;
+    public stiffNNonstiff: number = 0;
+    public lastError: number = 0;
+
+    // Stuff to tune
+    public maxSteps: number = 10000;
+    public atol: number = 1e-6;
+    public rtol: number = 1e-6;
+    public stiffCheck: number = 0;
+
+    constructor(rhs: RhsFn, n: number) {
+        this.stepper = new dopri5.Dopri5(rhs, n);
+        this.reset();
```

Comment:
Is this call to `reset` obsolete? The properties are initialised with the same values that this assigns